x-pack/plugins/enterprise_search/public/applications/enterprise_search/components/product_selector/product_selector.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext } from 'react';
+
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiPageContentBody,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { KibanaContext, IKibanaContext } from '../../../index';
+
+import { APP_SEARCH_PLUGIN, WORKPLACE_SEARCH_PLUGIN } from '../../../../../common/constants';
+
+import { SetEnterpriseSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { SendEnterpriseSearchTelemetry as SendTelemetry } from '../../../shared/telemetry';
+
+import { ProductCard } from '../product_card';
+
+import AppSearchImage from '../../assets/app_search.png';
+import WorkplaceSearchImage from '../../assets/workplace_search.png';
+
+interface IProductSelectorProps {
+  access: {
+    hasAppSearchAccess?: boolean;
+    hasWorkplaceSearchAccess?: boolean;
+  };
+}
+
+export const ProductSelector: React.FC<IProductSelectorProps> = ({ access }) => {
+  const { hasAppSearchAccess, hasWorkplaceSearchAccess } = access;
+  const {
+    config: { host },
+  } = useContext(KibanaContext) as IKibanaContext;
+
+  return (
+    <EuiPage restrictWidth className="enterpriseSearchOverview">
+      <SetPageChrome isRoot />
+      <SendTelemetry action="viewed" metric="overview" />
+
+      <EuiPageBody>
+        <EuiPageHeader>
+          <EuiPageHeaderSection className="enterpriseSearchOverview__header">
+            <EuiTitle size="l">
+              <h1 className="enterpriseSearchOverview__heading">
+                {i18n.translate('xpack.enterpriseSearch.overview.heading', {
+                  defaultMessage: 'Welcome to Elastic Enterprise Search',
+                })}
+              </h1>
+            </EuiTitle>
+            <EuiTitle size="s">
+              <p className="enterpriseSearchOverview__subheading">
+                {i18n.translate('xpack.enterpriseSearch.overview.subheading', {
+                  defaultMessage: 'Select a product to get started',
+                })}
+              </p>
+            </EuiTitle>
+          </EuiPageHeaderSection>
+        </EuiPageHeader>
+        <EuiPageContentBody>
+          <EuiFlexGroup justifyContent="center" gutterSize="xl">
+            {(!host || hasAppSearchAccess) && (
```

Comment:
Any thoughts if pulling this out into a var up above would make it more readable/easier to understand the business logic? 🤔 

something like:

```jsx
  const { hasAppSearchAccess, hasWorkplaceSearchAccess } = access;
  const { config } = useContext(KibanaContext) as IKibanaContext;

  // If Enterprise Search hasn't been set up yet, show all products. Otherwise, only show products the user has access to
  const shouldShowAppSearch = !config.host && hasAppSearchAccess;
  const shouldShowEnterpriseSearch = !config.host && hasWorkplaceSearchAccess;
```

and this line gets simplified to:

```suggestion
            {shouldShowAppSearch && (
```

LMK what you think/if you like it! The business logic behind the user flow wasn't super straightforward to me (as you can tell by the number of times I confused everyone w/ it) so personally I'd at least appreciate the extra comment that explains why/what the code's doing lol.

Reply:
Yeah, that occurred to me but thought it was overkill at the time. Happy to extract for better maintainability